package.json

```diff
@@ -72,6 +72,7 @@
     "test:unit:coverage": "jest --coverage",
     "test:unit:debug": "jest --debug",
     "test:unit:watch": "jest --watch",
-    "watch": "rollup --config ./scripts/rollup.config.js --watch"
+    "watch": "rollup --config ./scripts/rollup.config.js --watch",
+    "commit": "cz"
```

Comment:
cz is not included in the dependencies.

Reply:
@ninoish added it, [see here.](https://github.com/developerforce/lwc-builder-ui/pull/116/commits/5f4ad916e0aaf228e066b9564d6e3fe831b26052)

Reply:
@rahulgawale
It doesn't seem that Commitizen friendly settings has been made in the package.json and also please run npm install to include package-lock.json in the commit. 
https://github.com/commitizen/cz-cli#making-your-repo-commitizen-friendly

Reply:
@ninoish I have executed the below commands, respectively.

```
npm install commitizen -g
commitizen init cz-conventional-changelog --save-dev --save-exact

npm install --save-dev commitizen
npx commitizen init cz-conventional-changelog --save-dev --save-exact
```

Do I need to add the pre-commit hook as below to package.json?

```
"husky": {
  "hooks": {
    "prepare-commit-msg": "exec < /dev/tty && git cz --hook || true",
  }
}
```
Thanks

Reply:
Thanks. The husky configuration in package.json is the notation for the previous version of husky.
As we use husky v7, you need to add husky config under [.husky](https://github.com/developerforce/lwc-builder-ui/tree/main/.husky) folder.
This time, you need to add .husky/commit-msg file and include the script, and change the file permission to executable (+x).

https://typicode.github.io/husky/#/

Reply:
@ninoish I have updated the husky to the latest version but now, I am not able to commit because of these test failures.
1. 
```
Details:

    C:\Users\rahul.gawale\Documents\Personal Practice Orgs\lwc-builder\lwc-builder-ui\src\modules\my\buildContents\buildHtml.js:20
      if (targets?.lightning__RecordAction?.enabled && !targets?.lightning__RecordAction?.headlessAction) {
                  ^

    SyntaxError: Unexpected token '.'

       6 |  */
       7 |
    >  8 | import { buildHtml } from '../buildHtml';
         | ^
       9 |
      10 | describe('my-build-html', () => {
      11 |   afterEach(() => {

      at Runtime.createScriptFromCode (node_modules/jest-runtime/build/index.js:1350:14)
      at Object.<anonymous> (src/modules/my/buildContents/__tests__/buildHtml.test.js:8:1)
```


2.
 ```
Details:

    C:\Users\rahul.gawale\Documents\Personal Practice Orgs\lwc-builder\lwc-builder-ui\src\modules\my\app\app.js:37
        this.vscode?.postMessage(message);
                    ^

    SyntaxError: Unexpected token '.'

       7 |
       8 | import { createElement } from 'lwc';
    >  9 | import MyApp from 'my/app';
         | ^
      10 | import LWCBuilderEvent from 'domain/lwcBuilderEvent';
      11 |
      12 | // Mock vscode API

      at Runtime.createScriptFromCode (node_modules/jest-runtime/build/index.js:1350:14)
      at Object.<anonymous> (src/modules/my/app/__tests__/app.test.js:9:1)

PASS src/modules/my/form/__tests__/form.test.js
```
Can you guide me here?

Reply:
@rahulgawale 
You don't need to update husky version in this PR. Let's keep it simple.
Do you still get the same test failure with the current husky?

Reply:
@ninoish Yes, I reverted the husky version to the previous one, I am still getting these two errors. But these don't seem to be real errors, I think some config mismatch here. Also really sorry to drag this conversation, as I am pretty new to these things like husky, cz.

Reply:
@rahulgawale 
Ok, let's merge your main changes first, and leave the package.json config changes later.
Could you revert all changes in package.json and run npm install, then try committing again? If tests still fail, I'll look into it.

I knew cz thanks to you, and found it necessary to keep this project clean and comprehensive to everybody. So I'll definitely add it later. Thank you!

Reply:
@ninoish Thanks I have reverted all my commits until the first one, and I am still facing the test failures mentioned above. If you could merge this now, for now, I can fetch the upstream, for lateral changes, I think that should solve most of the issues I am facing.

Reply:
@rahulgawale
Ah, I think I found the reason. What version of Node.js do you use? The syntax that gets failure in the test is [introduced in v14](https://stackoverflow.com/questions/59574047/how-to-use-optional-chaining-in-node-js-12)). If you use v13 or below, is it ok to upgrade the version?

Reply:
@ninoish Updating the node version fixed the test suite issues now it is successfully getting passed. Thank you!

Reply:
@rahulgawale
Great, thanks! No more commits for this PR?

Reply:
@ninoish Not, for now, I will raise a new PR for the next changes.

Reply:
@rahulgawale 
Thank you for your contribution!